liquibase-core/src/main/java/liquibase/resource/PathHandlerFactory.java

```diff
@@ -75,17 +74,30 @@ public Resource getResource(String resourcePath) throws IOException {
      *
      * @return null if resourcePath does not exist and createIfNotExists is false
      * @throws IOException if there is an error opening the stream
+     *
+     * @deprecated use {@link #openResourceOutputStream(String, OpenOptions)}
      */
+    @Deprecated
     public OutputStream openResourceOutputStream(String resourcePath, boolean createIfNotExists) throws IOException {
+        return openResourceOutputStream(resourcePath, new OpenOptions().setCreateIfNeeded(createIfNotExists));
+    }
+
+    /**
+     * Returns the outputStream from {@link #getResource(String)}, using settings from the passed {@link OpenOptions}.
+     *
+     * @return null if resourcePath does not exist and {@link OpenOptions#isCreateIfNeeded()} is false
+     * @throws IOException if there is an error opening the stream
+     */
+    public OutputStream openResourceOutputStream(String resourcePath, OpenOptions openOptions) throws IOException {
         Resource resource = getResource(resourcePath);
         if (!resource.exists()) {
-            if (createIfNotExists) {
+            if (openOptions.isCreateIfNeeded()) {
```

Comment:
This needs a null check on openOptions

Reply:
This is identified as not null in the javadocs, and adding null checks everywhere it is used makes the code pretty complicated, so I'm not going to do this.